### PR TITLE
Remove old union literal syntax

### LINKED ISSUE #1175: Implement "Remove union literals from the language"
Standardized in https://github.com/dhall-lang/dhall-lang/pull/573.

I'm working on this!

### PR DESCRIPTION
**Pull request by sjakobi**
…as standardized in https://github.com/dhall-lang/dhall-lang/pull/573.

Fixes #1175.

TODO:

* [x] Complete parser changes (mostly names)
* [x] Fix docs for dhall-json
* [x] Fix Dhall.Tutorial and the tutorial tests
* [x] Fix error messages in Dhall.TypeCheck
* [x] Remove any other usage of the old syntax
* [x] Update the "Issue #108" benchmark

### CODE REVIEW

File: dhall/tests/Dhall/Test/Tutorial.hs [hunk 1]
```diff
@@ -33,7 +33,6 @@ tests =
         , Test.Tasty.testGroup "Unions"
             [ example 0 "./tests/tutorial/unions0A.dhall" "./tests/tutorial/unions0B.dhall"
             , example 1 "./tests/tutorial/unions1A.dhall" "./tests/tutorial/unions1B.dhall"
-            , example 2 "./tests/tutorial/unions2A.dhall" "./tests/tutorial/unions2B.dhall"
```
Comment (sjakobi) [inline dhall/tests/Dhall/Test/Tutorial.hs]: How come BTW that we have tests for the tutorial both here and in https://github.com/dhall-lang/dhall-lang/tree/master/tests/normalization/success/haskell-tutorial?

**Comment by sjakobi** [comment]
I discovered that the `issue108` benchmark was broken mostly by chance. That makes me wonder if any other benchmarks are broken too…

**Comment by sjakobi** [comment]
I also wonder why `dhall/benchmark/examples/issue108.dhall.bin` is now 110% larger. I should probably look into this!

**Comment by sjakobi** [comment]
> I also wonder why `dhall/benchmark/examples/issue108.dhall.bin` is now 110% larger. I should probably look into this!

Actually it's just a 10% size increase. Nothing to worry about.

**Comment by sjakobi** [comment]
Hydra complains about some breakage in `dhall-nix`. Is it intentional that `dhall-nix` isn't included in the stack.yaml?

File: dhall/tests/Dhall/Test/Tutorial.hs (hunk 1 above)
Comment (Gabriel439) [reply dhall/tests/Dhall/Test/Tutorial.hs]: No good reason :slightly_smiling_face:

**Comment by Gabriel439** [comment]
@sjakobi: I don't think there is any reason why `stack` doesn't build `dhall-nix`.  Most likely I just forgot to add it when merging `dhall-nix` into this repository

**Pull request merged**